src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -400,24 +415,34 @@ bool QgsGeoPackageConnectionItem::handleDrop( const QMimeData *data, Qt::DropAct
 
             QgsApplication::taskManager()->addTask( exportTask.release() );
           }
-        }
-        else
-        {
-          importResults.append( tr( "%1: Not a valid layer!" ).arg( u.name ) );
-          hasError = true;
-        }
-      }
-    }
-    else
-    {
-      // TODO: implement raster import
-      QgsMessageOutput *output = QgsMessageOutput::createMessageOutput();
-      output->setTitle( tr( "Import to GeoPackage database failed" ) );
-      output->setMessage( tr( "Failed to import some layers!\n\n" ) + QStringLiteral( "Raster import is not yet implemented!\n" ), QgsMessageOutput::MessageText );
-      output->showMessage();
-    }
+          else  // Import raster
+          {
+            // In case we need it
+            // QgsRasterLayer* rasterSrcLayer = dynamic_cast < QgsRasterLayer* > ( srcLayer );
+
+            const char *args[] = { "-of", "gpkg", "-co", QStringLiteral( "RASTER_TABLE=%1" ).arg( u.name ).toUtf8().constData(), "-co", "APPEND_SUBDATASET=YES", nullptr };
+            GDALTranslateOptions *psOptions = GDALTranslateOptionsNew( ( char ** )args, nullptr );
+            GDALDatasetH hSrcDS = GDALOpen( u.uri.toUtf8().constData(), GA_ReadOnly );
+            CPLErrorReset();
+            GDALDatasetH hOutDS = GDALTranslate( mPath.toUtf8().constData(), hSrcDS, psOptions, NULL );
```

Comment:
In the Debian/Ubuntu world, the jump was from 1.11 to 2.1, and people using for example Ubuntu 16.04 have to use UbuntuGIS to build QGIS that ship with at least 2.1 as well. OSGeo4W has GDAL 2.2. So would seem to be OK to me.